### PR TITLE
Update to nodejs rules 0.31.1 & bazel 0.26.0

### LINKED ISSUE #802: yarn_install & npm_install with managed directories can't handle deleted or manually regenerated node_modules folder
In nodejs rules 0.30.0, the yarn_install & npm_install rules now use the new managed directories Bazel 0.26.0 feature so the bazel build will use the user's node_modules folder (eliminating the need to install node_modules twice: once locally and once for the bazel build).

Originally, when the managed director feature was in design phase, it handled the case where node_modules was deleted or recreated manually (via yarn or npm) and the repository rule would re-run and everything would work. However, some of this design and scope of the feature changed and the repository rule is no longer run if node_modules is deleted or manually recreated. It only runs after a full `bazel clean --expunge` or when one of its inputs such as `package.json` or `yarn.lock` or `package-lock.json` changes.

This means that if you manually `rm -rf` your node_modules folder the bazel build will no longer work. You'll see an error that looks like this:

```
greg@macbook bazel_with_users_node_modules (master) $ bazel test ...
...
INFO: Found 1 test target...
Target //:main up-to-date:
  bazel-bin/main_bin.sh
  bazel-bin/main
INFO: Elapsed time: 16.640s, Critical Path: 2.44s
INFO: 2 processes: 2 darwin-sandbox.
INFO: Build completed successfully, 7 total actions
//:main                                                                  PASSED in 0.4s

Executed 1 out of 1 test: 1 test passes.
There were tests whose specified size is too big. Use the --test_verbose_timeout_warnings command line option to see which ones these arINFO: Build completed successfully, 7 total actions

greg@macbook bazel_with_users_node_modules (master) $ rm -rf node_modules

greg@macbook bazel_with_users_node_modules (master) $ bazel test ...
ERROR: /private/var/tmp/_bazel_greg/8cb9ee79286b251e74f070e4ed053074/external/npm/typescript/BUILD.bazel:123:1: no such package '@npm//node_modules/typescript': BUILD file not found on package path and referenced by '@npm//typescript:typescript'
ERROR: Analysis of target '//:main' failed; build aborted: Analysis failed
INFO: Elapsed time: 0.224s
INFO: 0 processes.
FAILED: Build did NOT complete successfully (0 packages loaded, 0 targets configured)
FAILED: Build did NOT complete successfully (0 packages loaded, 0 targets configured)
greg@macbook bazel_with_users_node_modules (master) $ 
```

Running `yarn` or `npm install` at that point still will not fix the build as the repository rule needs to rerun to create the npm fine grained build targets.

```
greg@macbook bazel_with_users_node_modules (master) $ yarn
yarn install v1.13.0
[1/4] 🔍  Resolving packages...
[2/4] 🚚  Fetching packages...
[3/4] 🔗  Linking dependencies...
[4/4] 🔨  Building fresh packages...
✨  Done in 0.34s.

greg@macbook bazel_with_users_node_modules (master) $ bazel test ...
ERROR: /private/var/tmp/_bazel_greg/8cb9ee79286b251e74f070e4ed053074/external/npm/typescript/BUILD.bazel:123:1: no such package '@npm//node_modules/typescript': BUILD file not found on package path and referenced by '@npm//typescript:typescript'
ERROR: Analysis of target '//:main' failed; build aborted: Analysis failed
INFO: Elapsed time: 0.210s
INFO: 0 processes.
FAILED: Build did NOT complete successfully (0 packages loaded, 0 targets configured)
FAILED: Build did NOT complete successfully (0 packages loaded, 0 targets configured)
```

The fix for this will likely come in Bazel 0.26.1 so this is mainly a tracking issue here.

### PR DESCRIPTION
**Pull request by gregmagolan**
* entry_point attribute of nodejs_binary & rollup_bundle is now a label
* symlinking of node_modules for yarn_install temporarily disabled for bazel schematics and the main yarn_install (not for integration/bazel) until the fix for https://github.com/bazelbuild/bazel/issues/8487 makes it into a future bazel release which will resolve https://github.com/bazelbuild/rules_nodejs/issues/802

### CODE REVIEW

File: packages/bazel/src/schematics/ng-add/index.ts [hunk 1]
```diff
@@ -48,10 +48,10 @@ function addDevDependenciesToPackageJson(options: Schema) {
 
     const devDependencies: {[k: string]: string} = {
       '@angular/bazel': angularCoreVersion,
-      '@bazel/bazel': '^0.26.0-rc.5',
+      '@bazel/bazel': '^0.26.0-rc13',
```
Comment (kyliau) [inline packages/bazel/src/schematics/ng-add/index.ts]: rc is not needed since 0.26.0 has been released?

**Review by kyliau — APPROVED** [review]

File: packages/bazel/src/schematics/ng-add/index.ts (hunk 1 above)
Comment (gregmagolan) [reply packages/bazel/src/schematics/ng-add/index.ts]: done. thanks.

**Comment by gregmagolan** [comment]
For reference, there was a managed directories flaky crash observed when running the bazel-schematics integration test locally. Issue file to Bazel here https://github.com/bazelbuild/bazel/issues/8488

**Comment by Toxicable** [comment]
Probably want bazel 0.26.1 with this which fixes managed directories being edited externally (by rm -rf  for example)

**Comment by alexeagle** [comment]
Thanks @Toxicable I hadn't seen that the fix was released. Just mirrored that to npm and will update this PR

**Review by gregmagolan — COMMENTED** [review]
LGTM. Thanks for finishing it up @alexeagle

**Review by IgorMinar — APPROVED** [review]
I love all the entry-point cleanup! this is so much better!

**Comment by IgorMinar** [comment]
mege-assistance: global approval

**Comment by gregmagolan** [comment]
Not quite done. Bazel 0.26.1 doesn't fix node_modules issue so need to put back `symlink_node_modules = False` again before merging.

**Comment by IgorMinar** [comment]
uh oh.. what's going on with node_modules? why's still broken?

**Comment by gregmagolan** [comment]
Scope change on the bazel side getting the managed_directories feature in led to an issue: https://github.com/bazelbuild/rules_nodejs/issues/802 with Bazel 0.26.0. Irina on Bazel team fixed it but hasn't landed in a release yet. Tracking for Bazel team here: https://github.com/bazelbuild/bazel/issues/8487

**Comment by gregmagolan** [comment]
`symlink_node_modules = False` back into main WORKSPACE and Bazel schematics. Ready to merge. These can be removed when Bazel is updated to a release which includes @irengrig fix https://github.com/bazelbuild/bazel/commit/6efc5b787ad3164cc2fb779c73377695032b4524

**Comment by Toxicable** [comment]
Ah my mistake, I assumed it was released when I saw the 0.26.1 release along with this being closed
https://github.com/bazelbuild/bazel/issues/8487 at very similar times

**Comment by IgorMinar** [comment]
@gregmagolan this one doesn't cleanly land on the patch branch, could you please send a separate PR that targets just 8.0.x? thanks

**Comment by gregmagolan** [comment]
> @gregmagolan this one doesn't cleanly land on the patch branch, could you please send a separate PR that targets just 8.0.x? thanks

Roger

**Pull request closed without merging**